src/hotspot/share/gc/serial/tenuredGeneration.cpp

```diff
@@ -499,7 +499,7 @@ void TenuredGeneration::complete_loaded_archive_space(MemRegion archive_space) {
   HeapWord* start = archive_space.start();
   while (start < archive_space.end()) {
     size_t word_size = cast_to_oop(start)->size();;
-    space->update_for_block(start, start + word_size);
+    _bts->update_for_block(start, start + word_size);
```

Comment:
The `update_for_block` is moved from `TenuredSpace` to `TenuredGeneration`. What about using `TenuredGeneration::update_for_block` here? It is also inlined by the compiler.

Reply:
Either is fine; I chose this way to make it explicit that this public api is called only by external users, not inside this class -- rather subjective.